templates/extras/calc.md

```diff
@@ -23,26 +23,30 @@ end
 ```
 In fact, `calc A = B : H ...` in tactic mode functions exactly like a call to `refine (calc A = B : H ...)`.
 
-## Error messages, and how to avoid them
+## Getting effective feedback while using calc
 
-Note that the error messages can be quite obscure when things aren't quite right, and often the red
-squiggles end up under a random `...`. A tip to avoid these problems with calc usage is to first
+To get helpful error messages, keep the calc structure even before the proof is complete. Use `_` as in the 
+example above or `sorry` to stand for missing justifications. `sorry` will supress error messages entirely, 
+while `_` will either generate a nice error message, or in tactic mode, a new subgoal. 
```

Comment:
Under the theory that "in tactic mode" is an adverbial phrase for "generate", I think the commas are supposed to go like this:
```suggestion
while `_` will either generate a nice error message or, in tactic mode, a new subgoal. 
```